OpenRA.Mods.Common/Activities/RepairBridge.cs

```diff
@@ -32,8 +34,13 @@ protected override void OnInside(Actor self)
 		{
 			if (hut.BridgeDamageState == DamageState.Undamaged || hut.Repairing || hut.Bridge.GetHut(0) == null || hut.Bridge.GetHut(1) == null)
 				return;
+
 			hut.Repair(self);
-			self.Dispose();
+
+			if (enterBehaviour == EnterBehaviour.Suicide)
+				self.Kill(self);
+			else if (enterBehaviour == EnterBehaviour.Dispose)
+				self.Dispose();
```

Comment:
Would it make sense to move that block into a method in the `Enter` activity? It's copied quite a few times, and afaics all those other activities inherit from `Enter`.